qiskit_experiments/curve_analysis/visualization/mpl_drawer.py

```diff
@@ -0,0 +1,361 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Curve drawer for matplotlib backend."""
+
+from typing import List, Dict, Sequence, Union, Optional, Callable
+
+import numpy as np
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+from matplotlib.ticker import ScalarFormatter, Formatter
+from qiskit.utils import detach_prefix
+from uncertainties import unumpy as unp, UFloat
+
+from qiskit_experiments.curve_analysis.curve_data import FitData
+from qiskit_experiments.framework import AnalysisResultData
+from qiskit_experiments.framework.matplotlib import get_non_gui_ax
+from .base_drawer import BaseCurveDrawer
+
+
+class MplCurveDrawer(BaseCurveDrawer):
+    """Curve drawer for MatplotLib backend."""
+
+    class PrefixFormatter(Formatter):
+        """Matplotlib axis formatter to detach prefix.
+
+        If axis value is x=1000 and the factor is 1000, it will be shown
+        to be 1.0 in the ticks and its unit is shown with the prefactor 'k'
+        in the axis label.
```

Comment:
```suggestion
        If a value is, e.g., x=1000.0 and the factor is 1000, then it will be shown
        as 1.0 in the ticks and its unit will be shown with the prefactor 'k'
        in the axis label.
```